sqlglot/dialects/mysql.py

```diff
@@ -243,6 +243,10 @@ class Parser(parser.Parser):
                 format=exp.Literal.string("%B"),
             ),
             "STR_TO_DATE": _str_to_date,
+            "TIMESTAMP": lambda args: exp.cast(
+                args[0],
```

Comment:
note: didn't use `seq_get` because it returns an optional type, and cast doesn't accept an optional

Reply:
you could avoid the cast then

exp.Cast(this=seq_get(), ...)

Reply:
I changed it up a bit after reading mysql docs